test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/Reader/JsonLight/RelativeUriReaderTests.cs

```diff
@@ -0,0 +1,588 @@
+
+                deltaWriter.WriteEnd();
+
+            }, requestUri, out contentType);
+
+            string FixReplacementFormat = "@odata.context\":\"{0}";
+            payload = startWithSlash
+                ? payload.Replace(string.Format(FixReplacementFormat, ServiceRoot), string.Format(FixReplacementFormat, "/SampleService/"))
+                : payload.Replace(string.Format(FixReplacementFormat, ServiceRoot), string.Format(FixReplacementFormat, ""));
+
+            output.WriteLine(payload);
+
+            this.ReadPayload(payload, "application/json", model, omReader =>
+            {
+                var deltaReader = omReader.CreateODataDeltaReader(et1Set, et1);
+                while (deltaReader.Read()) ;
+            }, requestUri);
+        }
+
+        private string WritePayload(EdmModel edmModel, Action<ODataMessageWriter> write, Uri requestUri, out string contentType)
+        {
+            var message = new InMemoryMessage() { Stream = new MemoryStream() };
+
+            var writerSettings = new ODataMessageWriterSettings
+            {
+                EnableMessageStreamDisposal = false
+            };
+
+            writerSettings.SetServiceDocumentUri(ServiceDocumentUri);
+            writerSettings.SetContentType(ODataFormat.Json);
+
+            if (requestUri != null)
+            {
+                ODataUriParser odataUriParser = new ODataUriParser(edmModel, ServiceDocumentUri, requestUri);
+                writerSettings.ODataUri = new ODataUri()
+                {
+                    ServiceRoot = ServiceDocumentUri,
+                    Path = odataUriParser.ParsePath()
+                };
+            }
+
+            using (var msgWriter = new ODataMessageWriter((IODataResponseMessage)message, writerSettings, edmModel))
+            {
+                write(msgWriter);
+            }
+
+            message.Stream.Seek(0, SeekOrigin.Begin);
+            using (StreamReader reader = new StreamReader(message.Stream))
+            {
+                contentType = message.GetHeader("Content-Type");
+                return reader.ReadToEnd();
+            }
+        }
+
+        private void ReadPayload(string payload, string contentType, EdmModel edmModel, Action<ODataMessageReader> test, Uri requestUri)
+        {
+            var message = new InMemoryMessage() { Stream = new MemoryStream(Encoding.UTF8.GetBytes(payload)) };
+            message.SetHeader("Content-Type", contentType);
+
+            ODataMessageReaderSettings readerSettings = new ODataMessageReaderSettings()
+            {
+                BaseUri = new Uri(ServiceDocumentUri, "$metadata"),
+                RequestUri = requestUri
+            };
+
+            using (var msgReader = new ODataMessageReader((IODataResponseMessage)message, readerSettings, edmModel))
+            {
+                test(msgReader);
+            }
+        }
+
+        private ODataResource CreateResource1(Uri id = null)

```

Comment:
Can we consolidate the CreateResource# into one method?

Reply:
done.

---
In reply to: [202220090](https://github.com/OData/odata.net/pull/1211#discussion_r202220090) [](ancestors = 202220090)